eng/tools/generate_api_report/src/main.rs

```diff
@@ -6,7 +6,26 @@ use std::io::prelude::*;
 use std::path::Path;
 use std::process::Command;
 
+// We pin to a specific nightly version to ensure compatibility with:
+// 1. The rust-api-parser tool (version 1.0.0) that consumes the rustdoc JSON output with FORMAT_VERSION (currently 37)
+// 2. The rustdoc_types crate version used in this tool (0.33.0)
+// When updating this version, ensure rustdoc_types dependency and the rust-api-parser tool are also updated.
+fn get_nightly_toolchain() -> Result<String, Box<dyn Error>> {
+    // Read the rust.yml file and extract the toolchain version
+    let yml_content = std::fs::read_to_string("eng/pipelines/templates/variables/rust.yml")?;
+
+    // Extract the value from the NIGHTLY_TOOLCHAIN_FOR_APIVIEW line
+    yml_content
+        .lines()
+        .find(|l| l.contains("NIGHTLY_TOOLCHAIN_FOR_APIVIEW"))
+        .and_then(|line| line.split('"').nth(1))
+        .map(|version| format!("+{}", version))
+        .ok_or_else(|| "Could not find NIGHTLY_TOOLCHAIN_FOR_APIVIEW in rust.yml".into())
+}
+
 fn main() -> Result<(), Box<dyn Error>> {
+    let nightly_toolchain = get_nightly_toolchain()?;
+
```

Comment:
Line 36 below is dangerous. You're assuming a PWD that might not be the case. It'll expect a `target` directory in PWD wherever the tool may be run.

How is this tool expected to run? I don't see anything in CI that actually runs it or how. Understanding the invocation is going to be important to know whether the PWD is right, etc. If run as `cargo run`, for example, `./target` will be wrong and `--package {name}` won't work because this project isn't in a workspace.

Reply:
Assumption is that the invocation is from the root.
some more context and a question - https://github.com/Azure/azure-sdk-for-rust/pull/2356/files#r2012970544

Reply:
Since I solved the PWD issue, probably no need but I would still update the `README.md` to mention that.

Reply:
readme has it already, threw in an error check too